package exception.checkedException.test;

import java.io.File;
import java.io.IOException;

public class CheckedExceptionTest {
	
	public static void main(String[] args) {
		
		try {
			criarArquivo();
		}catch(IOException e) {
			e.printStackTrace();
		}
	}
	
	//Melhor maneira para excecoes CHECKEDs:
	public static void criarArquivo() throws IOException{ //agora a responsabilidade de tratar a excecao é do metodo main, pq eu acho esse metodo nele!!!

		//IOException é uma classe que é checked (que deve ser tratada, nao compila senao tratar!)
		File file = new File("C:/projeto_java/exercicios/src/teste.txt");
		System.out.println("Arquivo criado: " + file.createNewFile());
		System.out.println("Arquivo criado");
	}
}

/* Exercicio anterior:
public static void criarArquivo() {
		//IOException é uma classe que é checked (que deve ser tratada, nao compila senao tratar!)
		File file = new File("C:/projeto_java/exercicios/src/teste.txt");
		
		try{
			System.out.println("Arquivo criado: " + file.createNewFile());
			System.out.println("Arquivo criado");
		} catch(IOException e) { // e é um objeto do tipo IOException
			e.printStackTrace(); //retorna uma pilha de métodos da exception
		}
	}
*/